### DISCUSSION TITLE
Private IPAM/DCIM as network-sources

### DISCUSSION THREAD

**Discussion by riddler63** [akvorado/akvorado#640] [Q&A]
Hello, I've installed akvorado (great tool btw) and want to use current IPAM (nautobot) as source to obtain prefix, tenant, role, site and region from it.

To get prefixes from nautobot the standard API request can be used:

HTTP GET `https://<nautobot-ip>/api/ipam/prefixes/`

The following jq expression can be used to represent data as required by akvorado, but the default output lacks region name.

`.results[] |
{prefix: .prefix, site: .site.name, tenant: .tenant.name, role: .role.name, }`

So, I've created simple GraphQL query (in Nautobot) and jq for it

GraphQL query
`query prefix_list {
  prefixes {
    prefix
    family
    tenant {
      name
    }
    role {
      name
    }
    site {
      name
      region {
        name
      }
    }
  }
}`
jq
`.data |
.prefixes[] |
{prefix: (.prefix), tenant: .tenant.name, role: .role.name, site: .site.name, region: .site.region.name}`

Unfortunately it requires HTTP POST method to run the GraphQL query, because one can define variables for the request:

HTTP POST `https://<nautobot-ip>/api/extras/graphql-queries/<graphql-query-id>/run/`

Also, both API requests require authentication by Token:

So, I have several questions:
1.  How I can define 'Authorization: Token' and other HTTP header options for the "url" in `network-sources:` configuration?
2. How I can specify HTTP POST method for the "url"?

**Comment by vincentbernat** [comment]
This has been implemented in 85561c44f7d0.
